### PR TITLE
Add documentation on result quality and confidence scores

### PR DESCRIPTION
**Pull request by orangejulius**
At the request of many users, this PR adds some information on confidence scores and other properties to help determine result quality.

A little bit of cleanup, especially in the organization of the table of contents, was done along the way.

Fixes https://github.com/pelias/documentation/issues/220

### CODE REVIEW

File: result_quality.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# Determining Result Quality
+
+Pelias tries very hard to always return good results, and generally does a pretty good job. If using the autocomplete endpoint or otherwise showing results directly to humans, judging result quality is generally easy (if subjective).
```
Comment (missinglink) [inline result_quality.md]: this line:

> If using the autocomplete endpoint or otherwise showing results directly to humans, judging result quality is generally easy (if subjective).

.. is quite long and hard to follow the meaning.

I read it as "Judging result quality is easy. This is especially true when using autocomplete, or alternatively when showing results directly to an end user.

.. but even written like that it doesn't really say much, I would suggest removing it, as eyeball congestion this early in the document might put people off?

File: result_quality.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+However, when programmatically managing results from Pelias, determining result quality is much harder. Pelias returns several different properties with each result to help determine if a result is good enough for your purposes.
```
Comment (missinglink) [inline result_quality.md]: each result *contains* several different properties?

how would you feel about adding 'air quotes' around things like 'good enough'?

File: result_quality.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+
+### layer
+This is essentially what type of result was returned. Examples are address, venue, neighbourhood, city, state. The `layers` parameter can be used to filter on this if you know ahead of time what you want. Results returned that are of a different layer than you expect generally mean incorrect results.
```
Comment (missinglink) [inline result_quality.md]: Could go for air-quotes again here for 'type'?

I would describe it more like:

"Layers are used to partition source data, they are analogous to a 'table' in a relational database model'

"In the Pelias model. layers generally define the 'granularity' of the result, from 'country' to 'neighbourhood' down to 'venue' and 'address'.

The `layers` parameter can be used to filter on this if you know ahead of time what you want.

Careful filtering of layers allows you to 'pick and choose' which results are returned.

**Comment by ashleyholt** [comment]
@orangejulius also wondering about the sentence that starts:

_"For forward geocoding and autocomplete, it's based on several factors such as as whether the result type was as expected ... "_

is this the same as saying:

_"For forward geocoding and autocomplete, it's based on several factors including the `match_type` ..."?_

(If not, what does it mean for the result type to be expected?)

Also wondering if it might be possible to add a table for forward geocoding and autocomplete to show progression of confidence scores, similar to what you have for reverse (realize this might be more challenging since there are multiple factors involved).

**Comment by orangejulius** [comment]
@ashleyholt yup, it's the same. I like your version better :)

A table would be harder for forward and autocomplete unfortunately.

**Comment by orangejulius** [comment]
@ashleyholt I reorganized the list of criteria to at least be a bulleted list, to better reflect that all of those considerations are weighted more or less equally. Let me know if that helps clear things up.
